### PR TITLE
crew: fix crew update reporting a potential package upgrade when a different package's package file is missing

### LINKED ISSUE #6254: Deleting a package file for an installed package confuses crew update
Simple test would be to install crew_launcher, and then do a crew update, which will then remove the `crew_launcher.rb` file since it hasn't been merged yet.

e.g.
```
export CREW_TESTING_BRANCH=crew-launcher
export CREW_TESTING_REPO=https://github.com/supechicken/chromebrew.git
CREW_TESTING=1 crew update
yes | CREW_TESTING=1 crew install crew_launcher
CREW_TESTING=1 crew upgrade crew_launcher
crew update
```

Now I see this:
```
chronos@localhost /usr/local/lib/crew/packages (master %|SPARSE=)$ crew update
From https://github.com/skycocker/chromebrew
 * branch            master     -> FETCH_HEAD
HEAD is now at c67ed9d Musl native toolchain + all packages necessary for static curl, git builds (#6062)
Package lists, crew, and library updated.
Checking for package updates...
pipewire could be updated from 0.3.35 to 0.3.29
audacious could be updated from 1 to 3.10.1
gcc11 could be updated from 11.2.0-1 to 11.1.0-2
libssp could be updated from 11.2.0 to 11.1.0
git could be updated from 2.33.0-musl to 2.32.0-musl-1
coreutils could be updated from 9.0 to 8.32

chronos@localhost /usr/local/lib/crew/packages (master %|SPARSE=)$ crew reinstall audacious ; crew update
audacious: Audacious is an open source audio player.
Performing pre-flight checks...
Precompiled binary available, downloading...
########################################################################################### 100.0%
Audacious archive downloaded.
Unpacking archive using 'tar', this may take a while...
Removing since upgrade or reinstall...
Audacious removed!
Performing pre-install...
Performing install...
Performing post-install for audacious...

To configure, execute the following:
Type 'audacious' to launch the music player.
From the menu, select 'Output' > 'Audio Settings'.
Select 'ALSA Output' for the 'Output plugin'.
Click 'Settings' to the right of 'ALSA Output'.
Select 'sysdefault - Default control device' for 'Mixer device'.
Click 'Close' and 'Close' again to save.

To completely remove, execute the following:
crew remove audacious audacious_plugins
rm -rf ~/.config/audacious

Audacious installed!
From https://github.com/skycocker/chromebrew
 * branch            master     -> FETCH_HEAD
HEAD is now at c67ed9d Musl native toolchain + all packages necessary for static curl, git builds (#6062)
Package lists, crew, and library updated.
Checking for package updates...
pipewire could be updated from 0.3.35 to 0.3.29
mpg123 could be updated from 1 to 1.25.10
gcc11 could be updated from 11.2.0-1 to 11.1.0-2
libssp could be updated from 11.2.0 to 11.1.0
git could be updated from 2.33.0-musl to 2.32.0-musl-1
coreutils could be updated from 9.0 to 8.32

Run `crew upgrade` to update all packages or `crew upgrade <package1> [<package2> ...]` to update specific packages.
```

Note how you see `audacious` in the crew update list, and then after reinstalling `audacious`, `mpg123` appears?

That's the bug...

@uberhacker @saltedcoffii @supechicken any ideas?

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #6254

- During `crew update`, avoid installed packages missing package files.
- This modifies the crew logic to set `@pkg` to nil in `search` if pkgPath can not be found. This avoids leaving @pkg set to the previous package searched if the current search doesn't find a package file. **This may have unintended consequences elsewhere as search is called from multiple places in `crew`.**

Works properly:
- [x] x86_64

### CODE REVIEW

File: bin/crew [hunk 1]
```diff
@@ -279,6 +279,10 @@ def search (pkgName, silent = false)
   rescue => e
     puts "Error with #{pkgName}.rb: #{e}".red unless e.to_s.include?('uninitialized constant')
```
Comment (uberhacker) [inline bin/crew:280]: Can we use `.lightred` instead of `.red`?  It adds clarity to the text.

File: bin/crew [hunk 2]
```diff
@@ -279,6 +279,10 @@ def search (pkgName, silent = false)
...
   end
+  if ! File.exist?(pkgPath) && silent
```
Comment (uberhacker) [inline bin/crew:282]: `unless File.exist?(pkgPath)`?

File: bin/crew (hunk 1 above)
Comment (satmandu) [reply bin/crew]: changed

File: bin/crew (hunk 2 above)
Comment (satmandu) [reply bin/crew]: changed

**Pull request merged**